project1-sps/script.js

```diff
@@ -1,4 +1,79 @@
+var NumPlayerWins = 0;
+var NumPlayerlosses = 0;
+var NumPlayerdraws = 0;
+
+var mode = "get user name";
+var UserName = "";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue;
+  if (mode == "get user name") {
+    //save username in global variable
+    UserName = input;
+    // change mode to play game
+    mode = "play game";
+    myOutputValue = `Hello ${UserName}`;
+  } else {
+    var playerInput = input;
+    var computerInput = getRandomObject();
+    console.log(computerInput);
+
+    var allWinningConditions =
+      (playerInput == SCISSORS && computerInput == PAPER) ||
+      (playerInput == STONE && computerInput == SCISSORS) ||
+      (playerInput == PAPER && computerInput == STONE);
+
+    var allLosingConditions =
+      (computerInput == SCISSORS && playerInput == PAPER) ||
+      (computerInput == STONE && playerInput == SCISSORS) ||
+      (computerInput == PAPER && playerInput == STONE);
+
+    var myOutputValue = "";
+
+    if (allWinningConditions) {
+      NumPlayerWins += 1;
+      myOutputValue = `Hello ${UserName}. <br> you win! <br> Total Wins ${NumPlayerWins}. Total Losses ${NumPlayerlosses}. Total Draws ${NumPlayerdraws}`;
+    } else if (allLosingConditions) {
+      NumPlayerlosses += 1;
+      myOutputValue = `Hello ${UserName}.  <br> you lose! <br>Total Wins ${NumPlayerWins}. Total Losses ${NumPlayerlosses}. Total Draws ${NumPlayerdraws}`;
+    } else {
+      NumPlayerdraws += 1;
+      myOutputValue = `Hello ${UserName}.  <br> draw. <br>Total Score ${NumPlayerWins}. Total Losses ${NumPlayerlosses}. Total Draws ${NumPlayerdraws}`;
+    }
+
+    // if computer beats player
+  }
+
   return myOutputValue;
 };
+
+//input = scissors / paper / stone
+// output =  win / lose
+// 6  cases = scissors tie / paper tie / stone tie/ scissors win paper/ paper win stone/ stone win scissors
+
+var SCISSORS = "scissors";
+var PAPER = "paper";
+var STONE = "stone";
+var REPLAY = "Type scissors, paper or stone to play again!";
```

Comment:
Good work defining these string constants with UPPERCASE! The reason we do this is because if we misspell the string "scissors" in the code, code sandbox won't show the red squiggly lines (because it's just a string of characters), but if we misspell the variable SCISSORS, codesandbox will know that the name is wrong and prompt us!